io/io/src/TBufferMerger.cxx

```diff
@@ -48,16 +46,13 @@ std::shared_ptr<TBufferMergerFile> TBufferMerger::GetFile()
 
 void TBufferMerger::Push(TBufferFile *buffer)
 {
-   {
-      std::lock_guard<std::mutex> lock(fQueueMutex);
-      fQueue.push(buffer);
-   }
+   std::lock_guard<std::mutex> lock(fQueueMutex);
```

Comment:
Please read the comment at http://en.cppreference.com/w/cpp/thread/condition_variable/notify_one.
It seems we don't need to hold the lock when notifying.